thunder/core/transform_common.py

```diff
@@ -142,20 +142,29 @@ def keep_or_swap(p):
 #   that only produce non-proxy objects
 # NOTE needed_proxies is an in/out argument, it takes an initial set of Variables you want to keep, and return
 #   all the needed proxies of the input trace
-def dce(trace: Trace, needed_proxies: None | set[Variable] = None) -> Trace:
+def dce(
+    trace_or_bsyms: Trace | list[BoundSymbolInterface], needed_proxies: None | set[Variable] = None, output=None
+) -> Trace | list[BoundSymbolInterface]:
```

Comment:
The function `dce` now accepts either a Trace or a list of BoundSymbolInterface objects, but the parameter output is required only in the latter case and lacks type annotation or validation.